docs/plugins.md

```diff
@@ -0,0 +1,269 @@
+Choose your Identity Management System and write custom authentication plugin as per detail mentioned in this section.
+
+
+> Currently, custom authenticators cannot be used to authenticate users of DataHub's web UI. This is because the DataHub web app expects the presence of 2 special cookies PLAY_SESSION and actor which are explicitly set by the server when a login action is performed.
+Instead, custom authenticators are useful for authenticating API requests to DataHub's backend (GMS), and can stand in addition to the default Authentication performed by DataHub, which is based on DataHub-minted access tokens.
+
+The sample authenticator implementation can be found at [Authenticator Sample](../metadata-service/plugin/src/test/sample-test-plugins)
+
+### Implementing an Authentication Plugin
+1. Add _datahub-auth-api_ as implementation dependency: Maven coordinates of _datahub-auth-api_ can be found at [Maven](https://mvnrepository.com/artifact/io.acryl/datahub-auth-api)
+
+   Example of gradle dependency is given below.
+
+   ```groovy
+    dependencies {
+        implementation 'io.acryl:datahub-auth-api:0.8.45'
+    }
+   ```
+2. Implement the Authenticator interface: Refer [Authenticator Sample](../metadata-service/plugin/src/test/sample-test-plugins)
+
+   <details>
+      <summary>Sample class which implements the Authenticator interface</summary>
+
+    ```java
+    public class GoogleAuthenticator implements Authenticator {
+   
+        @Override
+        public void init(@Nonnull Map<String, Object> authenticatorConfig, @Nullable AuthenticatorContext context) {
+          // Plugin initialization code will go here 
+          // DataHub will call this method on boot time
+        }
+   
+        @Nullable
+        @Override
+        public Authentication authenticate(@Nonnull AuthenticationRequest authenticationRequest)
+            throws AuthenticationException {
+            // DataHub will call this method whenever authentication decisions are need to be taken
+            // Authenticate the request and return Authentication
+        }
+    }
+   ```
+   </details>
+3. Bundle your Jar: Use `com.github.johnrengelman.shadow` gradle plugin to create an uber jar. 
+
+   To see an example of building an uber jar, check out the `build.gradle` file for the apache-ranger-plugin file of [Apache Ranger Plugin](../metadata-auth/apache-ranger-plugin) for reference. 
+
+   Exclude datahub plugin dependency and signature classes as shown in below `shadowJar` task.
+
+   ```groovy
+     apply plugin: 'com.github.johnrengelman.shadow';
+     shadowJar {
+         // Exclude com.datahub.plugins package and files related to jar signature   
+         exclude "com/linkedin/common/", "com/datahub/", "META-INF/*.RSA", "META-INF/*.SF","META-INF/*.DSA"
+     }
+   ```
+4. Refer section [Plugin Installation](#plugin-installation) for plugin installation in DataHub environment
+
+## Enable GMS Authentication
```

Comment:
Why is this section in this document? We can remove it.

Reply:
AuthenticationFilter.java has checked for this. if this environment variable is not set then it doesn't build the authentication chain. so I am referring to this section from `Plugin Installation`